static/js/navigation.js

```diff
@@ -0,0 +1,172 @@
+          }
+        }
+      } else {
+        dropdown.classList.remove('is-open');
+        dropdownContent.classList.add('u-hide');
+      }
+    });
+  });
+});
+
+// Close the menu if browser back button is clicked
+window.addEventListener('hashchange', function(event) {
+  navDropdowns.forEach(function (dropdown) {
+    const dropdownContent = document.getElementById(dropdown.id + "-content");
+
+    if (dropdown.classList.contains('is-open')) {
+      closeMenu(dropdown, dropdownContent);
+    }
+  });
+})
+
+dropdownWindowOverlay.addEventListener('click', function(event) {
+  navDropdowns.forEach(function(dropdown) {
+    var dropdownContent = document.getElementById(dropdown.id + "-content");
+
+    if (dropdown.classList.contains('is-open')) {
+      closeMenu(dropdown, dropdownContent);
+    }
+  });
+});
+
+function closeMenu(dropdown, dropdownContent) {
+  dropdown.classList.remove('is-open');
+  dropdownWindow.classList.add('slide-animation');
+  dropdownWindowOverlay.classList.add('fade-animation');
+  if (window.history.pushState) {
+    window.history.pushState(null, null, window.location.href.split('#')[0]);
+  }
+}
+
+if (window.location.hash) {
+  var tabId = window.location.hash.split('#')[1];
+  var tab = document.getElementById(tabId);
+  var tabContent = document.getElementById(tabId + '-content');
+
+  if (tab) {
+    setTimeout(function() {
+      document.getElementById(tabId).click();
+    }, 0);
+  }
+}
+
+function closeMainMenu() {
+  var navigationLinks = document.querySelectorAll('.p-navigation__dropdown-link');
+
+  navigationLinks.forEach(function(navLink) {
+    navLink.classList.remove('is-open');
+  });
+
+  if (!dropdownWindowOverlay.classList.contains('fade-animation')) {
+    dropdownWindowOverlay.classList.add('fade-animation');
+  }
+
+  if (!dropdownWindow.classList.contains('slide-animation')) {
+    dropdownWindow.classList.add('slide-animation');
+  }
+}
+
+var origin = window.location.href;
+
+addGANavEvents('#canonical-products', 'www.ubuntu.com-nav-0-products');
+addGANavEvents('#canonical-login', 'www.ubuntu.com-nav-0-login');
+addGANavEvents('#navigation', 'www.ubuntu.com-nav-1');
+addGANavEvents('#enterprise-content', 'www.ubuntu.com-nav-1-enterprise');
+addGANavEvents('#developer-content', 'www.ubuntu.com-nav-1-developer');
+addGANavEvents('#community-content', 'www.ubuntu.com-nav-1-community');
+addGANavEvents('#download-content', 'www.ubuntu.com-nav-1-download');
+addGANavEvents('.p-navigation--secondary', 'www.ubuntu.com-nav-2');
+addGANavEvents('.p-contextual-footer', 'www.ubuntu.com-footer-contextual');
+addGANavEvents('.p-footer__nav', 'www.ubuntu.com-nav-footer-0');
+addGANavEvents('.p-footer--secondary', 'www.ubuntu.com-nav-footer-1');
+
+function addGANavEvents(target, category){
+  var t = document.querySelector(target);
+  if (t) {
+    t.querySelectorAll('a').forEach(function(a) {
+      a.addEventListener('click', function(){
+        dataLayer.push({
```

Comment:
I get some console errors related to this when I open the navigation.
![image](https://user-images.githubusercontent.com/40214246/66774004-5c36d000-eeb8-11e9-937a-ded3aeee112a.png)

Reply:
I have added GTM so this will now work.